x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/store/reducer.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// eslint-disable-next-line import/no-nodejs-modules
+import { parse } from 'querystring';
```

Comment:
This is not good at all to pull this legacy code in for a few reasons.

The first, is the linter correctly went off to tell you _this is dangerous to bundle backend NodeJS code with front end code_

The second, is that if you look at the documentation for this utility:
https://nodejs.org/api/querystring.html

You will see that it is marked as `Stability: 3 - Legacy` and will be removed soon.

<img width="954" alt="Screen Shot 2021-09-29 at 10 24 47 AM" src="https://user-images.githubusercontent.com/1151048/135309833-b686c3ab-3fc0-4646-bad2-ad902df5a9ba.png">

I would very strongly consider using `URLSearchParams` and not adding this tech debt as suggested and `URLSearchParams` is native to both front and backend:
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

There are a lot of examples in the code base you can look for the usage of it.

I know we do this in a few other spots and we really shouldn't be, but we shouldn't be adding more.

Reply:
Thanks! I copy-pasted most of this builder code and could not figure why we were using the legacy module but didn't pay much attention to it. I'll replace it with URlSearchParams then.

Reply:
After checking the code, many types of many other parts of the code are relying on this module and its types to work.

I created an issue for our team to look into this problem https://github.com/elastic/security-team/issues/1786